expression/builtin_math_vec.go

```diff
@@ -857,11 +857,50 @@ func (b *builtinCeilDecToDecSig) vecEvalDecimal(input *chunk.Chunk, result *chun
 }
 
 func (b *builtinFloorDecToDecSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinFloorDecToDecSig) vecEvalDecimal(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDecimal, n)
```

Comment:
I think you don't need to allocate a new `buf` for this function. Both `buf` and `res` are Decimals, you can just reuse the `res` here.

Reply:
sorry, the type of buf is Column, but type of res is Decimal

Reply:
`buf` will be converted to `bufs` which is a slice of Decimal.
I mean you can use the `input` to replace the `buf`.

Reply:
OK，i'll amend it